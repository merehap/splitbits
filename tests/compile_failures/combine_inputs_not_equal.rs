use splitbits::*;

fn main() {
    let x: u8 = 0b1010_0101;
    let result = combinebits!(x, "aaaa aaaa bbbb bbbb");
    assert_eq!(result, 0b1010_0101_0000_1111u16);
}
