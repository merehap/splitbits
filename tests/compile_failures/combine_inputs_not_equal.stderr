error: proc macro panicked
 --> tests/compile_failures/combine_inputs_not_equal.rs:5:18
  |
5 |     let result = combinebits!(x, "aaaa aaaa bbbb bbbb");
  |                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = help: message: assertion `left == right` failed: The number of inputs must be equal to the number of names in the template.
            left: 1
           right: 2
